src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/SimpleRenameStrategy.cs

```diff
@@ -80,7 +76,7 @@ public override bool CanHandleRename(string oldFileName, string newFileName, boo
                 string newName = Path.GetFileNameWithoutExtension(newDocument.FilePath);
 
                 // Note that RenameSymbolAsync will return a new snapshot of solution.
-                renamedSolution = await _roslynServices.RenameSymbolAsync(newDocument.Project.Solution, symbol, newName).ConfigureAwait(false);
+                renamedSolution = await _roslynServices.RenameSymbolAsync(newDocument.Project.Solution, semanticModel.GetDeclaredSymbol(declaration), newName).ConfigureAwait(false);
```

Comment:
Ah I see - HasMatchingSyntaxNode would return if it was null and so we wouldnt get here?

---
In reply to: [112788275](https://github.com/dotnet/project-system/pull/2032#discussion_r112788275) [](ancestors = 112788275)